s3/pkg/service/object.go

```diff
@@ -304,6 +304,28 @@ func (s *s3Service) GetObject(ctx context.Context, req *pb.GetObjectInput, strea
 		return err
 	}
 
+	isAdmin, tenantId, err := util.GetCredentialFromCtx(ctx)
+	if err != nil && isAdmin == false {
+		log.Error("get tenant id failed")
+		err = ErrInternalError
+		return nil
+	}
+
+	if isAdmin == false {
+		switch object.Acl.CannedAcl {
+		case "bucket-owner-full-control":
+			if bucket.TenantId != tenantId {
+				err = ErrAccessDenied
+				return err
```

Comment:
Return nil or err?

Reply:
thanks for reviewing, I think that it is better to return err.

Reply:
Hi @sunfch If return err here, then what is getObjRes.ErrorCode used for?

Reply:
@sfzeng , It is also used to return the error info.

Reply:
We are checking this only for one of the canned ACL right? there are few other canned ACLs too right?

Reply:
@kumarashit,  thanks for your reviewing, other Canned ACL can be checked at default case.